src/marqo/tensor_search/tensor_search.py

```diff
@@ -1695,14 +1705,61 @@ def get_content_vector(possible_jobs: List[VectorisedJobPointer], job_to_vectors
                 raise not_found_error
     raise not_found_error
 
+def add_prefix_to_queries(queries: List[BulkSearchQueryEntity]) -> List[BulkSearchQueryEntity]:
+    prefixed_queries = []
+    for q in queries:
+        text_query_prefix = determine_text_prefix(q.text_query_prefix, q.index, "text_query_prefix")
 
-def run_vectorise_pipeline(config: Config, queries: List[BulkSearchQueryEntity], device: Union[Device, str]) -> Dict[
-    Qidx, List[float]]:
+        if q.q is None:
+            prefixed_q = q.q
+        elif isinstance(q.q, str):
+            if _is_image(q.q) and ((isinstance(q.index, UnstructuredMarqoIndex) and q.index.treat_urls_and_pointers_as_images) or
+                (isinstance(q.index, StructuredMarqoIndex))):
```

Comment:
We would have to check if the field is an `image_pointer` or not, if it's a structured index. If it's an image pointer, then it should have no prefix.

Reply:
I initially did, but I since this is for queries, I'm not sure that we need to - is there a way for users to define whether a query is an image pointer or not?

Reply:
I consulted Li on this, and he explained that all URLs pointing to images are treated as images and not text, so we can just simplify the condition to `if _is_image(q.q)`